### PR TITLE
Replay parsing

### PR DESCRIPTION
**Pull request by zachstence**

### CODE REVIEW

**Review by bwpikaard — APPROVED** [review]

File: src/sprocket-common/celery/celery.service.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+export class CeleryService {
+    private readonly logger = new Logger(CeleryService.name);
+
+    private readonly celeryClient = createClient(config.celery.broker, config.celery.backend);
+
+    private progressChannel: Channel;
+    
+    async onApplicationBootstrap(): Promise<void> {
+        this.logger.log(`Connecting to RabbitMQ @ ${config.celery.broker}`);
+        const connection = await connect(config.celery.broker);
```
Comment (0c370t) [inline src/sprocket-common/celery/celery.service.ts:26]: add `{heartbeat: 120}` as option for this call

File: src/sprocket-common/celery/celery.service.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * Starts a task synchronously (waits for execution to finish).
+     * @param task The task to execute.
+     * @param args Arguments for the specified task.
+     * @returns The return value from the task.
+     */
+    async runSync<T extends Task>(task: T, args: TaskArgs<T>): Promise<TaskResult<T>> {
+        const name = taskNames[task];
+
+        if (!Object.values(taskNames).includes(name)) {
```
Comment (0c370t) [inline src/sprocket-common/celery/celery.service.ts:41]: name being undefined would make more sense

File: src/sprocket-common/celery/celery.service.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            throw new Error(`Unsupported Celery task ${name}`);
```
Comment (0c370t) [inline src/sprocket-common/celery/celery.service.ts:42]: `name` -> `task`

File: src/sprocket-common/celery/celery.service.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+     * Starts a task asynchronously, not waiting for execution to finish. Returns the name of the queue
+     * where progress messages are published.
+     * @param task The task to execute.
+     * @param args Arguments for the specified task.
+     * @returns The name of the queue where task progress messages are published.
+     */
+    async run<T extends Task>(task: T, args: TaskArgs<T>): Promise<string> {
+        const name = taskNames[task];
+
+        if (!Object.values(taskNames).includes(name)) {
```
Comment (0c370t) [inline src/sprocket-common/celery/celery.service.ts:66]: same

File: src/sprocket-common/celery/celery.service.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+        const progressQueue = uuid();
+        await this.progressChannel.assertQueue(progressQueue, {exclusive: true});
+        args.progressQueue = progressQueue;
+
+        const t = this.celeryClient.createTask(name);
+        const asyncResult = t.applyAsync([], args);
+        this.logger.debug(`Running celery task asynchronously name=${name} taskId=${asyncResult.taskId}`);
+
+        asyncResult.get()
```
Comment (0c370t) [inline src/sprocket-common/celery/celery.service.ts:78]: Create a ticket to add support for a callback here

File: src/sprocket-common/sprocket-common.module.ts [hunk 6]
```diff
@@ -1,9 +1,11 @@
 import {Module} from "@nestjs/common";
 
 import {CachingModule} from "./caching";
+import {CeleryModule} from "./celery/celery.module";
 import {EventsModule} from "./events";
+import {MinioModule} from "./minio/minio.module";
 
 @Module({
-    imports: [CachingModule, EventsModule],
+    imports: [CachingModule, EventsModule, CeleryModule, CeleryModule, MinioModule],
```
Comment (0c370t) [inline src/sprocket-common/sprocket-common.module.ts:9]: Celery Module is here twice

**Review by 0c370t — CHANGES_REQUESTED** [review]
Mostly good; couple little changes

File: src/sprocket-common/sprocket-common.module.ts (hunk 6 above)
Comment (zachstence) [reply src/sprocket-common/sprocket-common.module.ts]: [f7130c8](https://github.com/SprocketBot/sprocket-common/pull/14/commits/f7130c8b45036d4ed6ada8ed5a199577ba49266d)

File: src/sprocket-common/celery/celery.service.ts (hunk 1 above)
Comment (zachstence) [reply src/sprocket-common/celery/celery.service.ts]: [a550227](https://github.com/SprocketBot/sprocket-common/pull/14/commits/a550227fed2b6b6d047d5e4ad43d654203810302)

File: src/sprocket-common/celery/celery.service.ts (hunk 4 above)
Comment (zachstence) [reply src/sprocket-common/celery/celery.service.ts]: [218519b](https://github.com/SprocketBot/sprocket-common/pull/14/commits/218519b13e5061b0a7eac878b31da9145df7b007)

File: src/sprocket-common/celery/celery.service.ts (hunk 3 above)
Comment (zachstence) [reply src/sprocket-common/celery/celery.service.ts]: [218519b](https://github.com/SprocketBot/sprocket-common/pull/14/commits/218519b13e5061b0a7eac878b31da9145df7b007)

File: src/sprocket-common/celery/celery.service.ts (hunk 2 above)
Comment (zachstence) [reply src/sprocket-common/celery/celery.service.ts]: [218519b](https://github.com/SprocketBot/sprocket-common/pull/14/commits/218519b13e5061b0a7eac878b31da9145df7b007)

File: src/sprocket-common/celery/celery.service.ts (hunk 5 above)
Comment (zachstence) [reply src/sprocket-common/celery/celery.service.ts:72]: [14d4b5b](https://github.com/SprocketBot/sprocket-common/pull/14/commits/14d4b5bb6b551ab916f4db38cd7c7d6cb6c109da)

**Pull request merged**